src/webview/notifications.ts

```diff
@@ -37,12 +37,22 @@ export const notificationsClassDefinition = `(() => {
     constructor(title = '', options = {}) {
       this.title = title;
       this.options = options;
-      window.ferdium.displayNotification(title, options)
-        .then(() => {
-          if (typeof (this.onClick) === 'function') {
-            this.onClick();
-          }
-        });
+      try{
+        window.ferdium.displayNotification(title, options)
+          .then(() => {
+            if (typeof (this.onClick) === 'function') {
+              this.onClick();
+            }
+          });
+      } catch(error) {
+	        this.options.onClick = null;
+          window.ferdium.displayNotification(title, options)
+            .then(() => {
+              if (typeof (this.onClick) === 'function') {
+                this.onClick();
+              }
+            });      
+      }
```

Comment:
Any reason we are having a try catch block instead of (async function).then().catch() ?

I agree everything is a bit of an hack and we should probably make some sense of it in the future.